packages/eas-cli/src/commands/env/unlink.ts

```diff
@@ -0,0 +1,97 @@
+import { Flags } from '@oclif/core';
+import chalk from 'chalk';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import { EASEnvironmentFlag, EASNonInteractiveFlag } from '../../commandUtils/flags';
+import { EnvironmentVariableScope } from '../../graphql/generated';
+import { EnvironmentVariableMutation } from '../../graphql/mutations/EnvironmentVariableMutation';
+import { EnvironmentVariablesQuery } from '../../graphql/queries/EnvironmentVariablesQuery';
+import Log from '../../log';
+import { getDisplayNameForProjectIdAsync } from '../../project/projectUtils';
+import { selectAsync } from '../../prompts';
+import { promptVariableEnvironmentAsync } from '../../utils/prompts';
+
+export default class EnvironmentVariableUnlink extends EasCommand {
+  static override description = 'link a shared environment variable to the current project';
+
+  static override hidden = true;
+
+  static override flags = {
+    name: Flags.string({
+      description: 'Name of the variable',
+    }),
+    ...EASEnvironmentFlag,
+    ...EASNonInteractiveFlag,
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectConfig,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    let {
+      flags: { name, 'non-interactive': nonInteractive, environment },
```

Comment:
we should verify that name and environment are specified for non-interactive mode